pagopa/io-app
# chore: [IOPID-167] TestAuthenticationScreen UI Refinement

## Short description
This PR:
- Handles all error situations with error redux action during the `test-login` API call.
- Adds `testLogin` volatile state to properly handle the UI state of `TestAuthenticationScreen`.
- Adds `TestAuthenticationScreen` tests.

<details><summary>Video</summary>
<p>

<video src="https://user-images.githubusercontent.com/16268789/234591117-f9dfbc4d-5483-4dd9-b0ff-03c674089848.mov" width="600" />

</p>
</details> 

## List of changes proposed in this pull request
- [ts/features/common/store/reducers/index.ts](https://github.com/pagopa/io-app/pull/4571/files#diff-b215994f68a368f1dc322f668b2937aeede527cbaa9242f10a9896526d1982aa): Adds the `TestLoginState` under `features` (not persisted, but volatile) to make it as encapsulated as possible.
- [ts/sagas/testLoginSaga.ts](https://github.com/pagopa/io-app/pull/4571/files#diff-7d599e5ee6dea19a2231e4a0e43ca1cc92d8d9680a2c6d7390e44999da2baf10): Adds missing `loginError` action dispatches.
- [ts/screens/authentication/TestAuthenticationScreen.tsx](https://github.com/pagopa/io-app/pull/4571/files#diff-54190256bc3cfec139b2b7e5e70c221f05d09fa8c9650ec95a3b38f6c9c36d91): Removes `native-base`, adds **error** and **success** UI states and **loading spinner**. Why a **success** state? To identify possible navigation issues.
- [ts/store/actions/authentication.ts](https://github.com/pagopa/io-app/pull/4571/files#diff-8f675a5e21b9a99d7ec5d3e2bb2cf950abb463a4e0e9525b5286035567ca87a8): Adds `testLoginCleanUp` action. This is useful to remove **error** and **success** views when you navigate back and forth during this states.
- [ts/store/reducers/testLogin.ts](https://github.com/pagopa/io-app/pull/4571/files#diff-6286d90afa4e0d1ffb5531596558295d753f7b1d0831d4d1e8448eef9d90fd32): Adds `testLoginReducer` and `testLoginSelector`.
- [ts/screens/authentication/__tests__/TestAuthenticationScreen.test.tsx](https://github.com/pagopa/io-app/pull/4571/files#diff-67816cf28cc48675238d98a2f9eb645014f1c8c4dd5cea88c7cb6478056a274b): Adds test login screen and store tests.

## How to test
Run the IO-APP against either production or io-dev-server and see what happens. If you run the app against the io-dev-server you could provide `error` and `delay` as password to test error and delayed responses.
